exec/java-exec/src/test/java/org/apache/drill/test/TestGracefulShutdown.java

```diff
@@ -179,40 +180,28 @@ public void testRestApiShutdown() throws Exception {
   }
 
   @Test // DRILL-6912
-  public void gracefulShutdownThreadShouldBeInitializedBeforeClosingDrillbit() throws Exception {
-    Drillbit drillbit = null;
-    Drillbit drillbitWithSamePort = null;
-
-    int userPort = QueryTestUtil.getFreePortNumber(31170, 300);
-    int bitPort = QueryTestUtil.getFreePortNumber(31180, 300);
+  public void testDrillbitWithSamePortContainsShutdownThread() throws Exception {
     ClusterFixtureBuilder fixtureBuilder = ClusterFixture.bareBuilder(dirTestWatcher).withLocalZk()
-        .configProperty(ExecConstants.INITIAL_USER_PORT, userPort)
-        .configProperty(ExecConstants.INITIAL_BIT_PORT, bitPort);
-    try (ClusterFixture clusterFixture = fixtureBuilder.build()) {
-      drillbit = clusterFixture.drillbit();
-
-      // creating another drillbit instance using same config
-      drillbitWithSamePort = new Drillbit(clusterFixture.config(), fixtureBuilder.configBuilder().getDefinitions(),
-          clusterFixture.serviceSet());
-
+        .configProperty(ExecConstants.ALLOW_LOOPBACK_ADDRESS_BINDING, true)
+        .configProperty(ExecConstants.INITIAL_USER_PORT, QueryTestUtil.getFreePortNumber(31170, 300))
+        .configProperty(ExecConstants.INITIAL_BIT_PORT, QueryTestUtil.getFreePortNumber(31180, 300));
+
+    try (ClusterFixture fixture = fixtureBuilder.build();
+         Drillbit drillbitWithSamePort = new Drillbit(fixture.config(),
+             fixtureBuilder.configBuilder().getDefinitions(), fixture.serviceSet())) {
+      // Assert preconditions :
+      //      1. First drillbit instance should be started normally
+      //      2. Second instance startup should fail, because ports are occupied by the first one
+      assertNotNull("First drillbit instance should be initialized", fixture.drillbit());
```

Comment:
Looks better, drillbit received from `fixture.drillbit()` will be closed during closing `ClusterFixture`.

Reply:
Yes, I looked at close() method in fixture.